### PR TITLE
Full rework on Mail system

### PR DESCRIPTION
**Pull request by csoler**
Needs PR 2658 from Retroshare/ to test. Please read the instructions and warning there.

### CODE REVIEW

File: src/retroshare/rsmsgs.h [hunk 1]
```diff
@@ -150,13 +163,28 @@ class MsgAddress
 		explicit MsgAddress(const std::string& email, MsgAddress::AddressMode mmode)
 			: _type(MSG_ADDRESS_TYPE_EMAIL), _mode(mmode), _addr_string(email){}
```
Comment (G10h4ck) [inline src/retroshare/rsmsgs.h:164]: `MSG_ADDRESS_TYPE_EMAIL` at some point should be rename into something that doesn't make confusion with actual email ( myname@mydoman.tld )

File: src/retroshare/rsmsgs.h [hunk 2]
```diff
@@ -150,13 +163,28 @@ class MsgAddress
...
+        RsGxsId toGxsId()     const { checkType(MSG_ADDRESS_TYPE_RSGXSID);return RsGxsId (_addr_string);}
+        RsPeerId toRsPeerId() const { checkType(MSG_ADDRESS_TYPE_RSPEERID);return RsPeerId(_addr_string);}
+        std::string toEmail() const { checkType(MSG_ADDRESS_TYPE_EMAIL   );return          _addr_string ;}
+        std::string toStdString() const { return _addr_string ;}
+
+        void clear() { _addr_string=""; _type=MSG_ADDRESS_TYPE_UNKNOWN; _mode=MSG_ADDRESS_MODE_UNKNOWN;}
+        void checkType(MsgAddress::AddressType t) const
+        {
+            if(t != _type)
+                RsErr() << "WRONG TYPE in MsgAddress. This is not a good sign. Something's going wrong." ;
```
Comment (G10h4ck) [inline src/retroshare/rsmsgs.h:185]: Better using `RS_ERR` so you get also the function name prepended "for free"

File: src/serialiser/rstypeserializer.h [hunk 3]
```diff
@@ -143,7 +143,7 @@ struct RsTypeSerializer
 					print_stacktrace();
 					break;
 				}
-				memcpy(&member, ctx.mData + ctx.mOffset, sizeof(INTT));
+                memcpy(&member, ctx.mData + ctx.mOffset, sizeof(INTT));
```
Comment (G10h4ck) [inline src/serialiser/rstypeserializer.h:146]: This line is only a white space editing?

File: src/services/broadcastdiscoveryservice.cc [hunk 4]
```diff
@@ -262,6 +262,9 @@ BroadcastDiscoveryService::createResult(
 	 * any case should not be treated here */
 	auto bdp = BroadcastDiscoveryPack::fromSerializedString(uData, ec);
 
+    if(!bdp)
```
Comment (G10h4ck) [inline src/services/broadcastdiscoveryservice.cc:265]: This seems ended up here spuriously, this fix is already included in master AFAIK

File: src/serialiser/rstypeserializer.h [hunk 5]
```diff
@@ -515,7 +524,7 @@ struct RsTypeSerializer
 				        << " elCount: " << elCount
 				        << " ctx.mSize: " << ctx.mSize
 				        << " ctx.mOffset: " << ctx.mOffset << " "
-				        << std::errc::argument_out_of_domain
+                        << std::errc::argument_out_of_domain
```
Comment (G10h4ck) [inline src/serialiser/rstypeserializer.h:527]: white space editing here too, is your Qt creator using the correct profile?

**Review by G10h4ck — APPROVED** [review]
I suggests a few minor cleanups inline, but it is good for me :-)

File: src/retroshare/rsmsgs.h (hunk 1 above)
Comment (csoler) [reply src/retroshare/rsmsgs.h]: done

**Pull request merged**